semeio/workflows/localisation/local_script_lib.py

```diff
@@ -236,21 +236,24 @@ def write_scaling_values(
         raise KeyError(f" Method name: {method_name} is not implemented")
 
     scaling_values = np.zeros((nx, ny, nz), dtype=np.float32)
-    for k, j, i in itertools.product(nz, ny, nx):
-        global_index = grid_for_field.global_index(ijk=(i, j, k))
-        scaling_values[i, j, k] = decay_obj(global_index)
+    for k in range(nz):
+        for j in range(ny):
```

Comment:
This was just a typo on my end, it should be `itertools.product(range(nz), range(ny), range(nx))`.

The fact that this was not discovered in the testing is worrying me a bit, can we add a test covering it?

Reply:
Any reason why using itertools ? It is more readable as a triple loop. The reason why it was not testet was that it dependet on a test where LogLevel was set to LEVEL4 and that was not included, but I will remove the dependency on LogLevel for the write scaling to file functionality and then it is natural to have a test where this is activated by a keyword in the config file.

Reply:
The reason for `itertools` is that the triple loop does not really add anything as you are not really doing anything in loops 1 and 2, the idea is that you are looping through a grid, and not the individual layers. Great if you add a test, you can also call the function directly, so you do not depend on the log level for it to run.

Reply:
Are there any performance gain by using itertools instead of triple loop? In that case I can change this back and use itertools,.  It is used in the function writing calculated scaling factor and that functionality does not any longer depend on log level, only on whether the user has turned on or off writing this to file using a new keyword in the config file. In the updated test_integration I have a test where the calculated scaling factor is written to file.